### PR TITLE
Introduce org.entangle_photo.Manager application

### PR DESCRIPTION
**Pull request by berrange**
This is derived from work first started by Manuel Quiñones in #1171

With the changes made on top of that proposal, this is flatpak
is able to successfully connect to a camera and shoot photos.
In addition the Python plugins load successfully.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

### CODE REVIEW

File: org.entangle_photo.Manager.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
+    "app-id": "org.entangle_photo.Manager",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.34",
```
Comment (Erick555) [inline org.entangle_photo.Manager.json:4]: Latest runtime version is 3.36

Comment (hfiguiere) [reply org.entangle_photo.Manager.json:4]: you beat me to it ;-)

**Comment by hfiguiere** [comment]
bot, build org.entangle_photo.Manager

File: org.entangle_photo.Manager.json (hunk 1 above)
Comment (berrange) [reply org.entangle_photo.Manager.json]: Refreshed this to 3.36

**Comment by LeandroStanger** [comment]
bot, build org.entangle_photo.Manager

File: org.entangle_photo.Manager.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        "--socket=wayland",
+        "--filesystem=host",
+        "--metadata=X-DConf=migrate-path=/org/entangle-photo/manager/",
+        /* Needed for gvfs to work */
+        "--talk-name=org.gtk.vfs",
+        "--talk-name=org.gtk.vfs.*",
+        /* For USB, until there is a better way in flatpak */
+        "--device=all"
+    ],
+    "modules": [
```
Comment (barthalion) [inline org.entangle_photo.Manager.json:23]: I've noticed some static libraries and headers lying around. You can add a global cleanup array, `["*.a", "*.la", "/lib/pkgconfig", "/lib/cmake", "/include"]` should be good enough.

File: org.entangle_photo.Manager.json [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    "command": "entangle",
+    "rename-icon": "entangle",
+    "rename-desktop-file": "entangle.desktop",
+    "rename-appdata-file": "entangle.appdata.xml",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=host",
```
[the remark this answers is not in the document]
Comment (berrange) [reply org.entangle_photo.Manager.json:15]: Possibly, I'll check this.

File: org.entangle_photo.Manager.json (hunk 2 above)
Comment (berrange) [reply org.entangle_photo.Manager.json:23]: Oh yes, that's reasonabe.

Comment (hfiguiere) [reply org.entangle_photo.Manager.json:23]: maybe it is time we have libgphoto2 in `shared-modules`... off the top of my head this is the 4th app that uses it. And I'm sure there are more.

Comment (berrange) [reply org.entangle_photo.Manager.json:23]: Yeah, in fact I feel that applies to all the deps I have there IMHO. Every single one of them is in use by at least one other application I know of that's based on GTK/GNOME.

Comment (bilelmoussaoui) [reply org.entangle_photo.Manager.json:23]: It's not a complicated to build dependency, we don't  want shared-modules to become a new dependency place thing. Apps are supposed to bundle the necessary libraries that aren't available in the SDK. Shared-modules is mostly for complicated, very common ones

**Comment by berrange** [comment]
This update has fixed a broken icon in the help about box, cleaned up unnecessary files, and restricted filesystem access

**Comment by bilelmoussaoui** [comment]
bot, build org.entangle_photo.Manager

**Pull request closed without merging**